txircd/ircd.py

```diff
@@ -385,7 +375,150 @@ def reloadModule(self, moduleName):
+		if "server_description" not in config:
+			raise ConfigValidationError("server_description", "required item not found in configuration file.")
+		if not isinstance(config["server_description"], basestring):
+			raise ConfigValidationError("server_description", "value must be a string")
+		if "network_name" not in config:
+			raise ConfigValidationError("network_name", "required item not found in configuration file.")
+		if not isinstance(config["network_name"], basestring):
+			raise ConfigValidationError("network_name", "value must be a string")
+		if len(config["network_name"]) > 32:
+			config["network_name"] = config["network_name"][:32]
+			self.logConfigValidationWarning("network_name", "value is too long", config["network_name"])
+		if not "bind_client" not in config:
+			config["bind_client"] = [ "tcp:6667:interface={::}" ]
+			self.logConfigValidationWarning("bind_client", "no default client binding specified", "[ \"tcp:6667:interface={::}\" ]")
+		if not isinstance(config["bind_client"], list):
+			raise ConfigValidationError("bind_client", "value must be a list")
+		for bindDesc in config["bind_client"]:
+			if not isinstance(bindDesc, basestring):
+				raise ConfigValidationError("bind_client", "every entry must be a string")
+		if "bind_server" not in config:
+			config["bind_server"] = []
+		if not isinstance(config["bind_server"], list):
+			raise ConfigValidationError("bind_server", "value must be a list")
+		for bindDesc in config["bind_server"]:
+			if not isinstance(bindDesc, basestring):
+				raise ConfigValidationError("bind_server", "every entry must be a string")
+		if "modules" not in config:
+			config["modules"] = []
+		if not isinstance(config["modules"], list):
+			raise ConfigValidationError("modules", "value must be a list")
+		for module in config["modules"]:
+			if not isinstance(module, basestring):
+				raise ConfigValidationError("modules", "every entry must be a string")
+		if "links" in config:
+			if not isinstance(config["links"], dict):
+				raise ConfigValidationError("links", "value must be a dictionary")
+			for desc, server in config["links"].iteritems():
+				if not isinstance(desc, basestring):
+					raise ConfigValidationError("links", "\"{}\" is an invalid server description".format(desc))
+				if not isinstance(server, dict):
+					raise ConfigValidationError("links", "values for \"{}\" must be a dictionary".format(desc))
+				if "connect_descriptor" not in server:
+					raise ConfigValidationError("links", "server \"{}\" must contain a \"connect_descriptor\" value".format(desc))
+				if "in_password" in server:
+					if not isinstance(server["in_password"], basestring):
+						config["links"][desc]["in_password"] = str(server["in_password"])
+				if "out_password" in server:
+					if not isinstance(server["out_password"], basestring):
+						config["links"][desc]["out_password"] = str(server["out_password"])
+		if "datastore_path" not in config:
+			config["datastore_path"] = "data.db"
```

Comment:
We can also remove the `.get` call for this if we're just going to default it here.